File: 이재훈/프로그래머스/Lv1/폰켓몬.py
def solution(nums):
    answer = 0
    nums_di = {}
    for i in nums:
        if i in nums_di.keys():
            nums_di[i]+=1
        else:
            nums_di[i]=1
    
    answer=min(len(nums_di.keys()), len(nums)//2)
    return answer

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.*;

class Solution {
    public int solution(int[] nums) {
        int answer = 0;
        int select = nums.length/2;
        HashMap<String, Boolean> map = new  HashMap<>();
        
        for(int i : nums) map.put(String.valueOf(i), true);
        
        if(map.size() > select) return select;
        else answer = map.size();
        
        return answer;
    }
}